xunit-autolabeler-v2/ast_parser/core/polyglot_drift_data.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from typing import List, NamedTuple, Tuple
+
+
+class PolyglotDriftData(NamedTuple):
+    """Struct for storing snippet metadata
+    This object stores language-agnostic ("polyglot")
+    snippet data extracted from snippet source files
+    for use by the second-stage "polyglot" parser.
+    """
+    source_path: str
+    start_line: int
+    end_line: int
+    parser: str
+    region_tags: List[str] = []
+    test_methods: List[Tuple[str, str]] = []
+    children: List[str] = []
+
+    @classmethod
+    def from_json_dict(cls, json_dict):
+        region_tags = json_dict.get('region_tags', [])
+        start_line = json_dict.get('start_line', None)
+        end_line = json_dict.get('end_line', None)
+        test_methods = json_dict.get('test_methods', [])
+        source_path = json_dict.get('source_path', [])
+        children = json_dict.get('children', [])
+        parser = json_dict.get('parser', None)
+
+        return cls(
+            source_path,
+            start_line,
+            end_line,
+            parser,
+            region_tags,
+            test_methods,
+            children
+        )
```

Comment:
You can actually pass a Dict as kwargs to fill up the Namedtuple, that way you don't need to maintain this extra classmethod.

polyglot_drift_data = PolyglotDriftData(**my_dict)